pkg/virt-operator/kubevirt.go

```diff
@@ -820,18 +824,23 @@ func (c *KubeVirtController) garbageCollectInstallStrategyJobs() error {
 }
 
 func (c *KubeVirtController) getInstallStrategyFromMap(config *operatorutil.KubeVirtDeploymentConfig, generation int64) (*install.Strategy, bool) {
-	c.installStrategyMutex.Lock()
-	defer c.installStrategyMutex.Unlock()
+	cachedValue := c.latestStrategy.Load()
+	if cachedValue == nil {
+		return nil, false
+	}
+	cachedEntry, ok := cachedValue.(strategyCacheEntry)
+	if !ok {
+		return nil, ok
+	}
 
-	strategy, ok := c.installStrategyMap[fmt.Sprintf(installStrategyKeyTemplate, config.GetDeploymentID(), generation)]
-	return strategy, ok
+	if cachedEntry.key == fmt.Sprintf(installStrategyKeyTemplate, config.GetDeploymentID(), generation) {
+		return cachedEntry.value, true
+	}
+	return nil, false
 }
 
-func (c *KubeVirtController) cacheInstallStrategyInMap(strategy *install.Strategy, config *operatorutil.KubeVirtDeploymentConfig, generation int64) {
-
-	c.installStrategyMutex.Lock()
-	defer c.installStrategyMutex.Unlock()
-	c.installStrategyMap[fmt.Sprintf(installStrategyKeyTemplate, config.GetDeploymentID(), generation)] = strategy
+func (c *KubeVirtController) cacheInstallStrategyInMap(cachedEntry *install.Strategy, config *operatorutil.KubeVirtDeploymentConfig, generation int64) {
```

Comment:
nit: maybe we can also change the function name?

Reply:
See the latest commit.